gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/poles_equipped.mcfunction

```diff
@@ -0,0 +1,6 @@
+# Give jump boost effects when holding poles, as long as the player is on a block (to simulate a cooldown)
+# @s = player holding poles
+# at @s
+# run from main
+
+execute at @a unless block ~ ~-0.2 ~ minecraft:air run effect give @s jump_boost 1 0 true
```

Comment:
The `@a` here creates issues, you probably meant `@s`; regardless, do this location setting in `main`.
Creating a cooldown via the `unless block air` check is possible, however, only with high clock speeds, which nullifies the cooldown effect. If you want a cooldown, implement one using (e.g.) scores.